pkg/cloudprovider/drift.go

```diff
@@ -25,16 +25,26 @@ import (
 	"github.com/aws/karpenter-core/pkg/cloudprovider"
 	"github.com/aws/karpenter/pkg/apis/v1alpha1"
 	"github.com/aws/karpenter/pkg/providers/amifamily"
+	"github.com/aws/karpenter/pkg/providers/instance"
 	"github.com/aws/karpenter/pkg/utils"
 )
 
 func (c *CloudProvider) isNodeTemplateDrifted(ctx context.Context, machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
+	ec2Instance, err := c.getInstance(ctx, machine.Status.ProviderID)
+	if err != nil {
+		return false, cloudprovider.IgnoreMachineNotFoundError(err)
+	}
+
 	amiDrifted, err := c.isAMIDrifted(ctx, machine, provisioner, nodeTemplate)
 	if err != nil {
 		return false, cloudprovider.IgnoreMachineNotFoundError(fmt.Errorf("calculating ami drift, %w", err))
 	}
+	securitygroupDrifted := c.isSecurityGroupDrifted(ec2Instance, nodeTemplate)
```

Comment:
```suggestion
	securitygroupDrifted := c.areSecurityGroupsDrifted(ec2Instance, nodeTemplate)
```

nit: It's plural since we are validating that all security groups are propagated to the instance